patsy/util.py

```diff
@@ -550,3 +562,127 @@ def test_iterable():
+    # Categorical object, and Categorical objects are what have .categories /
+    # .codes attributes.
+    if hasattr(cat, "cat"):
+        cat = cat.cat
+    if hasattr(cat, "categories"):
+        return cat.categories
+    else:
+        return cat.levels
+
+# Needed to support pandas < 0.15
+def pandas_Categorical_codes(cat):
+    # In 0.15+, a categorical Series has a .cat attribute which is a
+    # Categorical object, and Categorical objects are what have .categories /
+    # .codes attributes.
+    if hasattr(cat, "cat"):
+        cat = cat.cat
+    if hasattr(cat, "codes"):
+        return cat.codes
+    else:
+        return cat.labels
+
+def test_pandas_Categorical_accessors():
+    c = pandas_Categorical_from_codes([1, 1, 0, -1], ["a", "b"])
+    assert np.all(pandas_Categorical_categories(c) == ["a", "b"])
+    assert np.all(pandas_Categorical_codes(c) == [1, 1, 0, -1])
+
+    if have_pandas_categorical_dtype:
+        s = pandas.Series(c)
+        assert np.all(pandas_Categorical_categories(s) == ["a", "b"])
+        assert np.all(pandas_Categorical_codes(s) == [1, 1, 0, -1])
+
+# Needed to support pandas >= 0.15 (!)
+def safe_is_pandas_categorical_dtype(dt):
+    if not have_pandas_categorical_dtype:
+        return False
+    # WTF this incredibly crucial function is not even publically exported.
+    # Also if you read its source it uses a bare except: block which is broken
+    # by definition, but oh well there is not much I can do about this.
+    return pandas.core.common.is_categorical_dtype(dt)
+
+# Needed to support pandas >= 0.15 (!)
+def safe_is_pandas_categorical(data):
+    if not have_pandas_categorical:
+        return False
+    if isinstance(data, pandas.Categorical):
+        return True
+    if hasattr(data, "dtype"):
+        return safe_is_pandas_categorical_dtype(data.dtype)
+    return False
+
+def test_safe_is_pandas_categorical():
+    assert not safe_is_pandas_categorical(np.arange(10))
+
+    if have_pandas_categorical:
+        c_obj = pandas.Categorical.from_array(["a", "b"])
+        assert safe_is_pandas_categorical(c_obj)
+
+    if have_pandas_categorical_dtype:
+        s_obj = pandas.Series(["a", "b"], dtype="category")
+        assert safe_is_pandas_categorical(s_obj)
+
+# Needed to support pandas >= 0.15 (!)
+# Calling np.issubdtype on a pandas categorical will blow up -- the officially
+# recommended solution is to replace every piece of code like
+#   np.issubdtype(foo.dtype, bool)
+# with code like
+#   isinstance(foo.dtype, np.dtype) and np.issubdtype(foo.dtype, bool)
+# or
+#   not pandas.is_categorical_dtype(foo.dtype) and issubdtype(foo.dtype, bool)
+# We do the latter (with extra hoops) because the isinstance check is not
+# safe. See
+#   https://github.com/pydata/pandas/issues/9581
+#   https://github.com/pydata/pandas/issues/9581#issuecomment-77099564
+def safe_issubdtype(dt1, dt2):
+    if safe_is_pandas_categorical_dtype(dt1):
+        return False
```

Comment:
There is no such logic possible AFAIK. There's no way to "subdtype" pandas categorical (esp. b/c it isn't a dtype), and there is no `issub...` API for checking it either.